### ISSUE TITLE
Overfitting when loading the pretrained model to train own dataset

### ISSUE THREAD

**Issue by zknus** [megvii-research/AnchorDETR#12]
Hi, thanks for your great work. 
When I load the anchor_detr_r50_dc5.pth into the model and training on other dataset. The training loss is reducing, but the test loss remains around the same value in the 30 epoches. Do you have any experience with this? I reduced the number of reference points and the number of patterns, but still the same problem.

**Comment by tangjiuqi097** [comment]
@zknus Hi, the small MLP network is not used to update the position of the points. It is treated as the position encoding function that maps a point coordinate to a position embedding vector. We think the heuristic position encoding function is not optimal and can not well make use of the position of the anchor points, thus we use a small MLP additionally. The learned position encoding function can improve about 1~2 AP than the heuristic function.

**Comment by zknus** [comment]
@tangjiuqi097   Hi, thanks for your prompt reply. 
So in my understanding, you are using the sine position encoding function and learned position encoding function, right? https://github.com/megvii-research/AnchorDETR/blob/6274fca833bc521c915e344963c05e6248e2976e/models/transformer.py#L297-L298  But the MLP can learn and update the weight of the position embedding, why can't we say that the anchor points' position is also learned and updated?

**Comment by zknus** [comment]
@tangjiuqi097 Hi, yes the figure 1 in your paper shows very nice results of your anchor points. It shows that the anchor points can give the position reference for the object. However, from your code, https://github.com/megvii-research/AnchorDETR/blob/6274fca833bc521c915e344963c05e6248e2976e/models/transformer.py#L61-L62 and https://github.com/megvii-research/AnchorDETR/blob/6274fca833bc521c915e344963c05e6248e2976e/models/transformer.py#L106-L107 , it seems that the learned anchor points still use the same way defined in DETR, which is https://github.com/facebookresearch/detr/blob/091a817eca74b8b97e35e4531c1c39f89fbe38eb/models/detr.py#L39.   
This is my question that what's the difference in the implematation part. 
Here is my understanding, the Object Queries in the DETR learns the object information from the whole image set, while your anchor points are initialized in each batch size, the samll MLP network is learned and updated for the whole image dataset to represent the learned object position. Please correct me if I'm wrong. Thanks.

**Comment by tangjiuqi097** [comment]
@zknus  hi,

The `nn.Embedding` is used to generate the learnable tensor. You can also use the implementation of `nn.Parameter(torch.Tensor(self.num_position, 2))` instead.  The learnable parameters have different meanings.


BTW, the following understanding seems not right.
> anchor points are initialized in each batch size

**Comment by zknus** [comment]
@tangjiuqi097 Hi, yes, both of them are learnable tensor. Could you please explain more about this "The learnable parameters have different meanings." ?  In my understanding, the anchor points and the Object Queries in DETR are basically  defined in the same way but with different training strategies. The anchor points are also added in the final prediction of the bounding box in your code.  Thanks.

**Comment by tangjiuqi097** [comment]
@zknus hi, the object query in DETR does not have an explicit physical meaning. It may be related to the position, scale, and some other properties of the objects. But we do not know what it actually is. 

On the contrary, we get the anchor point first and encode the coordinate (x,y) of the anchor point as the object query. So the object query represents the position encoding of the anchor point. You can also refer to the encoding of the key feature position and the fixed grid anchor point, which I think may help you understand.
